overlord/devicestate/devicemgr.go

```diff
@@ -357,11 +424,106 @@ func getSerial(t *state.Task, privKey asserts.PrivateKey, device *auth.DeviceSta
 	return serial, nil
 }
 
+type serialRequestConfig struct {
+	requestIDURL     string
+	serialRequestURL string
+	headers          map[string]string
+	proposedSerial   string
+	body             []byte
+}
+
+func (cfg *serialRequestConfig) applyHeaders(req *http.Request) {
+	for k, v := range cfg.headers {
+		req.Header.Set(k, v)
+	}
+}
+
+func getFromTransaction(tr *configstate.Transaction, gadgetName, key string, v interface{}) error {
```

Comment:
I suggest moving this into Transaction itself as `GetMaybe` or similar, which takes a default value as the last option.